### DISCUSSION TITLE
I'm confused: what is a segment, and what is a line?

### DISCUSSION THREAD

**Discussion by MattPenaTV** [openai/whisper#1868] [Q&A]
From "whisper --help":

```
  --max_line_width MAX_LINE_WIDTH
                        (requires --word_timestamps True) the maximum number of characters in a line before breaking the
                        line (default: None)
  --max_line_count MAX_LINE_COUNT
                        (requires --word_timestamps True) the maximum number of lines in a segment (default: None)
  --max_words_per_line MAX_WORDS_PER_LINE
                        (requires --word_timestamps True, no effect with --max_line_width) the maximum number of words
                        in a segment (default: None)

```
Based on these definitions, I find it hard to understand what qualifies as a "segment" or a "line". max_line_count refers to "the maximum number of lines in a segment", making it clear that lines and segments are two different things... or so I thought. Because max_words_per_line is defined as "the maximum number of words in a **segment**", implying that a segment and a line are the same thing. So what's the deal? Are a segment and a line the same thing, or two different things? What is a segment, and what is a line? Is there an error in the "--help" here or am I just misinterpreting something?

Cheers

**Comment by ryanheise** [comment]
I think the documentation above for `--max_words_per_line MAX_WORDS_PER_LINE` is incorrect and should say "the maximum number of words in a line".

So a segment is the block of text that you can see on the screen at once in a single subtitle. By default that all appears in one line, but you can make it wrap over multiple lines by setting the maximum number of characters in a line or the maximum number of words in a line, and then setting a limit for how many lines you want to allow within the subtitle block before spilling over into a new subtitle block.

**Reply by MattPenaTV** [reply]
So a subtitle block is considered a "segment", and these "segments" are composed of lines. Am I getting that right?

<img width="228" alt="Screen Shot 2023-12-03 at 10 21 15 AM" src="https://github.com/openai/whisper/assets/107569942/d69cfe6e-a012-4880-86c2-4206d7262bc2">

Meaning that each of these subtitle blocks (1,2,3,4) is considered a "segment"?


<img width="282" alt="Screen Shot 2023-12-03 at 10 34 16 AM" src="https://github.com/openai/whisper/assets/107569942/7735939b-3df8-461a-a354-306ddbaf8dd8">

And each indentation in the "segment" is a "line"

**Reply by ryanheise** [reply]
All correct except maybe the word indentation is unclear (to a programmer indentation means spaces or tabs at the start of a line, of which there are none here). But a line is probably what you think it is: a single horizontal row of text that spans horizontally but not vertically. So in your second example, each segment is wrapped over multiple lines.

**Reply by MattPenaTV** [reply]
So for each line should start with indentation?

**Reply by ryanheise** [reply]
I was just suggesting that we avoid the word indentation because it means something different to you and me.

This is a line.
This is another line.

Above there are two lines because there are two horizontal rows of text.

**Reply by MattPenaTV** [reply]
That makes perfect sense.

If you mean avoiding word indentation on lines, like this:

This is an unindented line.
    This is an indented line.

Then I totally agree. If not, I'm not sure what you mean by "word indentation". Could you please elaborate?

**Reply by ryanheise** [reply]
By "avoid the word", I mean let's not say that word because the word is unclear and confusing, and what you think that word means is different from what a programmer typically thinks that word means, so we can't communicate with each other when we are using two different meanings of that word. That is why I gave my explanation without using that word.
